services/compute/virtual-machines/taxonomy.md

```diff
@@ -0,0 +1,47 @@
+|	CCC-030115	|	Instance Lifecycle Events	|	Offering features for managing the lifecycle and state of virtual machine instances, including starting, stopping, pausing, and restarting instances as needed.	|
+|	CCC-030116	|	Instance Snapshots	|	Creation of snapshots of virtual machine instances to capture and preserve state and data for backup and cloning purposes. 	|
+|	CCC-030117	|	Instance Templates	|	Offering templates for provisioning virtual machine instances with pre-configured images, predefined configurations and settings and/or bootstrap scripts.	|
+|	CCC-030118	|	Instance Preemptibility	|	Providing the option for using preemptible virtual machine (spot) instances at a lower cost for non-critical or fault-tolerant workloads that may be terminated by the cloud provider after a notice period.	|
+|	CCC-030119	|	Instance Affinity/Anti-affinity	|	Enabling control over the location of virtual machine instances to ensure or prevent co-location on the same physical hardware.	|
+|	CCC-030120	|	Instance Health Checks	|	Providing features for performing health checks on virtual machine instances and automatically replacing or repairing unhealthy instances.	|
+|	CCC-030121	|	Instance Remote Access	|	Offering remote access to virtual machine instances through methods such as SSH or RDP for troubleshooting, debugging, and maintenance purposes.	|
+|	CCC-030122	|	Instance Live Migration	|	The ability to perform live migration of virtual machine instances between physical hosts for maintenance or load balancing purposes without downtime.	|
+|	CCC-030123	|	Instance Remote Configuration	|	Providing tools for remotely configuring virtual machine instances, including deployment automation and configuration management frameworks.	|
+|	CCC-030124	|	Instance Resource Tagging	|	Enabling tagging of virtual machine instances with metadata for organization, management, and cost allocation purposes.	|
+|	CCC-030125	|	Instance Resource Monitoring	|	Providing tools for monitoring resource utilization and performance metrics for virtual machine instances, including CPU usage, memory usage, disk I/O, and network traffic.	|
+|	CCC-030126	|	Custom Images	|	Allows users to create and manage their own customized virtual machine images.	|
+|	CCC-030127	|	Dynamic Performance	|	Providing "burstable" instances for intermittent workloads that accumulate credits during periods of low usage which can be used to burst above baseline performance when needed.	|
+|	CCC-030128	|	Dedicated Instances	|	Providing the option to run instances on physical servers that are dedicated solely to a single customer account, ensuring that the underlying hardware resources are not shared with other customers.	|
+|	CCC-030129	|	Instance Scheduling	|	Offering features for scheduling the startup, shutdown, and maintenance of virtual machine instances based on predefined schedules or conditions.	|
```

Comment:
My knowledge is not very sharp on this. Please correct me if I'm wrong. Start/Stop instance base on a schedule should live in a higher level like in a separate Scheduler service. Start/Stop scripts on a instance is part of cron jobs (I assume we can still have a VM which does not support cron jobs and still be able to call it a VM). Maintenance base on schedule should be a feature of a external patching service, etc. (not a feature of VM). Just my thought

Reply:
Poked around a bit more to check these and I agree that this scheduling is largely handled by other services. There's an argument to be made with GCP that the instance groups are part of the VM solution (they can be used to establish schedules as well as auto-scaling and grouping), but we have another thread on that issue above for people to weigh in on whether those belong in this taxonomy. For now I'll remove this feature, but I'll back up the markdown somewhere in case we want it back.